### ISSUE TITLE
Quickfix 'Remove the redundant modifier.' also removes javadoc.

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-xtend#415]

**Comment by cdietrich** [comment]
@meysholdt any ideas?

**Comment by kthoms** [comment]
@miklossy @cdietrich Can this be closed?

**Comment by miklossy** [comment]
No. As far as I know, the bug still exists, see the [Ignored QuickfixTest test cases](https://github.com/eclipse/xtext-xtend/blob/1ad4e80835e9c8c0b1628ee46793c8dde3859b66/org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend#L4080)

**Comment by kthoms** [comment]
Thanks for the update. Any chance to fix this for 2.14 or shall we retarget it?

**Comment by miklossy** [comment]
retarget

**Comment by cdietrich** [comment]
main problem is that i dont have the time to analyze this right now and @meysholdt seems to have neither

**Comment by kthoms** [comment]
That's OK, it is a minor issue from my perspective that's not mandatory for 2.14. So let's fix it in 2.15.

**Comment by cdietrich** [comment]
@meysholdt 

`HiddenRegionPartAssociator.associate(IHiddenRegion, BiConsumer<IHiddenRegionPart, HiddenRegionPartAssociation>)` classifies the comment as 
`HiddenRegionPartAssociation.NEXT` 

i dont understand why it is then removed if there are parts for the xtendclass (the semantic element) left.
can you please a least give my a tiny hint where to look at?

**Comment by cdietrich** [comment]
same for comments https://github.com/eclipse/xtext/issues/1360

**Comment by cdietrich** [comment]
does not work in all cases unfortunately.

**Comment by cdietrich** [comment]
@meysholdt some hints really would help

**Comment by miklossy** [comment]
Actually I wonder why it is working on fields without any explicit type, see test case [unnecessaryModifier_23](https://github.com/eclipse/xtext-xtend/blob/0ebf99074744bc61b295f32635688c7bde944732/org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend#L4687)
```
class Foo {
	/**
	 * javadoc
	 */
	f|inal val a = 1
}
```

and why it is not working on fields with explicit type, see test case [unnecessaryModifier_20](https://github.com/eclipse/xtext-xtend/blob/0ebf99074744bc61b295f32635688c7bde944732/org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend#L4618)
```
class Foo {
	/**
	 * javadoc
	 */
	pri|vate int a = 10
}
```

I think it would be a good starting point to debug these two test cases and identify the differences. That could lead us into the right direction.

**Comment by cdietrich** [comment]
as a workaround i will provide a possibility to define textual multi quickfixes

**Comment by cdietrich** [comment]
Workaround inplace for 217
